app/services/workflow_delete_service.rb

```diff
@@ -0,0 +1,23 @@
+class WorkflowDeleteService
+  attr_accessor :workflow
+
+  def initialize(workflow_id)
+    self.workflow = Workflow.find(workflow_id)
+  end
+
+  def cleanup
```

Comment:
just called `delete`

Reply:
also need to delete the entries in tag links table, but we also need to delete the actual tag remotely.

Reply:
Name is changed into `delete`. The local tag links will be cleaned up automatically. Remote tags need to implement `DeleteRemoteTags` service first. Marked `TODO` here, will add it later.